st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -32,23 +33,110 @@
+    'ActionExecutionsController'
+]
 
 LOG = logging.getLogger(__name__)
 
-
 MONITOR_THREAD_EMPTY_Q_SLEEP_TIME = 5
 MONITOR_THREAD_NO_WORKERS_SLEEP_TIME = 1
 
 
-class ActionExecutionsController(ResourceController):
+class ActionExecutionsControllerMixin(RestController):
     """
-        Implements the RESTful web endpoint that handles
-        the lifecycle of ActionExecutions in the system.
+    Mixin class with shared methods.
     """
+
     model = ActionExecutionAPI
     access = ActionExecution
+
+    # A list of attributes which can be specified using ?exclude_attributes filter
+    valid_exclude_attributes = [
+        'result',
+        'trigger_instance'
+    ]
+
+    def _get_result_object(self, id):
+        """
+        Retrieve result object for the provided action execution.
+
+        :param id: Action execution ID.
+        :type id: ``str``
+
+        :rtype: ``dict``
+        """
+        fields = ['result']
+        action_exec_db = self.access.impl.model.objects.filter(id=id).only(*fields).get()
+        return action_exec_db.result
+
+    def _get_children(self, id_, depth=-1, result_fmt=None):
+        # make sure depth is int. Url encoding will make it a string and needs to
+        # be converted back in that case.
+        depth = int(depth)
+        LOG.debug('retrieving children for id: %s with depth: %s', id_, depth)
+        descendants = execution_service.get_descendants(actionexecution_id=id_,
+                                                        descendant_depth=depth,
+                                                        result_fmt=result_fmt)
+        return [self.model.from_model(descendant) for descendant in descendants]
+
+    def _validate_exclude_fields(self, exclude_fields):
+        """
+        Validate that provided exclude fields are valid.
+        """
+        if not exclude_fields:
+            return exclude_fields
+
+        for field in exclude_fields:
+            if field not in self.valid_exclude_attributes:
+                msg = 'Invalid or unsupported attribute specified: %s' % (field)
+                raise ValueError(msg)
+
+        return exclude_fields
+
+
+class ActionExecutionChildrenController(ActionExecutionsControllerMixin):
+    @jsexpose(str)
+    def get(self, id, **kwargs):
+        """
+        Retrieve children for the provided action execution.
+
+        :rtype: ``list``
+        """
+        return self._get_children(id_=id, **kwargs)
+
+
+class ActionExecutionAttributeController(ActionExecutionsControllerMixin):
+    @jsexpose()
+    def get(self, id, attribute, **kwargs):
+        """
+        Retrieve a particular attribute for the provided action execution.
+
+        Handles requests:
+
+            GET /actionexecutions/<id>/attribute/<value>
+
+        :rtype: ``dict``
+        """
+        fields = [attribute]
+        fields = self._validate_exclude_fields(fields)
+        action_exec_db = self.access.impl.model.objects.filter(id=id).only(*fields).get()
```

Comment:
Also, we should aim to address issues in other PR's and not in this one otherwise 0.8.1 will diverge from master which is bad.

And looks like merging 0.8.1 into master will be kinda painful as well.